java/src/main/java/com/powsybl/dataframe/network/adders/BatteryDataframeAdder.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.dataframe.network.adders;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.dataframe.SeriesMetadata;
+import com.powsybl.dataframe.update.UpdatingDataframe;
+import com.powsybl.iidm.network.BatteryAdder;
+import com.powsybl.iidm.network.Network;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Yichen TANG <yichen.tang at rte-france.com>
+ * @author Etienne Lesot <etienne.lesot at rte-france.com>
+ * @author Sylvain Leclerc <sylvain.leclerc@rte-france.com>
+ */
+public class BatteryDataframeAdder extends AbstractSimpleAdder {
+
+    private static final List<SeriesMetadata> METADATA = List.of(
+            SeriesMetadata.stringIndex("id"),
+            SeriesMetadata.strings("voltage_level_id"),
+            SeriesMetadata.strings("bus_id"),
+            SeriesMetadata.strings("connectable_bus_id"),
+            SeriesMetadata.ints("node"),
+            SeriesMetadata.strings("name"),
+            SeriesMetadata.doubles("max_p"),
+            SeriesMetadata.doubles("min_p"),
+            SeriesMetadata.doubles("p0"),
+            SeriesMetadata.doubles("q0")
+    );
+
+    @Override
+    public List<List<SeriesMetadata>> getMetadata() {
+        return Collections.singletonList(METADATA);
+    }
+
+    @Override
+    public void addElement(Network network, UpdatingDataframe dataframe, int indexElement) {
+        BatteryAdder batteryAdder = network.getVoltageLevel(dataframe.getStringValue("voltage_level_id", indexElement)
+                        .orElseThrow(() -> new PowsyblException("voltage_level_id is missing")))
+                .newBattery();
+        NetworkElementCreationUtils.createInjection(batteryAdder, dataframe, indexElement);
```

Comment:
Be careful here: if not present, I think that you will have NaN values.

Reply:
See discussion synthesis about default values in comment below.